packages/core/src/agent/Agent.ts

```diff
@@ -194,13 +194,13 @@ export class Agent {
       transport.stop()
     }
 
-    // close/delete wallet if still initialized
+    // close wallet if still initialized
     if (this.wallet.isInitialized) {
-      if (deleteWallet) {
-        await this.wallet.delete()
-      } else {
-        await this.wallet.close()
-      }
+      await this.wallet.close()
+    }
+    // delete wallet if it has been created
+    if (this.wallet.isCreated && deleteWallet) {
```

Comment:
I think we should throw an error here, or update the config parameter to include that it will only delete if the wallet is provisioned.

Or maybe remove the option? It is mostly used for tests, but now that the wallet api is public we can just use that right?

Reply:
I kind of like that `shutdown` method, also deleting a wallet if you need it. But you're right. It does not make sense in the "real" world outside the test. We need to delete the wallet manually, which is likely worth the effort.